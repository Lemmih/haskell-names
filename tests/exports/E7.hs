module E7 (Foo(..), Bar(..), N(..)) where

data Foo = Foo1 | Foo2 Int | Foo3 { c :: Bool }

class Bar where x :: Foo

newtype N = N { unN :: Foo }

a = a

data Baz = Baz
