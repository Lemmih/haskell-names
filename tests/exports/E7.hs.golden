Symbols (
  fromList
    [ SymMethod
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "x"
                          }
        , sv_fixity = Nothing
        , sv_className = OrigName
                           { origPackage = Nothing
                           , origGName = GName
                                           "E7"
                                           "Bar"
                           }
        }
    , SymSelector
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "c"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo"
                          }
        }
    , SymSelector
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "unN"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "N"
                          }
        }
    , SymConstructor
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo1"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo"
                          }
        }
    , SymConstructor
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo2"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo"
                          }
        }
    , SymConstructor
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo3"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo"
                          }
        }
    , SymConstructor
        { sv_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "N"
                          }
        , sv_fixity = Nothing
        , sv_typeName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "N"
                          }
        }
    ] ) (
  fromList
    [ SymData
        { st_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Foo"
                          }
        , st_fixity = Nothing
        }
    , SymNewType
        { st_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "N"
                          }
        , st_fixity = Nothing
        }
    , SymClass
        { st_origName = OrigName
                          { origPackage = Nothing
                          , origGName = GName
                                          "E7"
                                          "Bar"
                          }
        , st_fixity = Nothing
        }
    ] )